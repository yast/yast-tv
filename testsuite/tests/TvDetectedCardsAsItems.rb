# encoding: utf-8

module Yast
  class TvDetectedCardsAsItemsClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"
      # testing with different values of detected_cards
      TEST_It_Now(nil)

      TEST_It_Now(
        [
          {
            "module"        => "bttv",
            "name"          => "First detected card",
            "unique_key"    => "hluk.HvsIesmXXX1",
            "sound_card_no" => 0
          }
        ]
      )

      TEST_It_Now(
        [
          {
            "module"     => "xyz",
            "name"       => "My TV_card",
            "unique_key" => "j7F3.J8dhgs34fK2"
          },
          {
            "module"        => "bttv",
            "name"          => "Some other detected card",
            "unique_key"    => "hluk.YYY2",
            "sound_card_no" => 0
          }
        ]
      )

      nil
    end

    def TEST_It_Now(detected)
      detected = deep_copy(detected)
      Tv.detected_cards = deep_copy(detected)
      DUMP("---------------")

      # Tv::cards (... already installed cards) == nil
      Tv.cards = nil
      detected_items = Tv.DetectedCardsAsItems
      DUMP(detected_items)

      # Tv::cards == $[]
      Tv.cards = []
      detected_items = Tv.DetectedCardsAsItems
      DUMP(detected_items)

      # Tv::cards == card1
      Tv.cards = [
        {
          "module"        => "bttv",
          "parameters"    => { "param1" => "testvalue" },
          "unique_key"    => "hluk.HvsIesmXXX1",
          "sound_card_no" => 0
        }
      ]
      detected_items = Tv.DetectedCardsAsItems
      DUMP(detected_items)

      # Tv::cards == card1, nil, nil, card4
      Tv.cards = [
        {
          "module"        => "bttv",
          "parameters"    => { "param1" => "testvalue" },
          "unique_key"    => "hluk.HvsIesmXXX1",
          "sound_card_no" => 0
        },
        nil,
        nil,
        {
          "module"        => "xyz",
          "parameters"    => { "par_other" => "testvalue2" },
          "name"          => "My TV_card",
          "unique_key"    => "j7F3.J8dhgs34fK2",
          "sound_card_no" => 1
        }
      ]
      detected_items = Tv.DetectedCardsAsItems
      DUMP(detected_items)

      nil
    end
  end
end

Yast::TvDetectedCardsAsItemsClient.new.main
