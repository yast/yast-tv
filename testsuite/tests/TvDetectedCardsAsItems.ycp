{

    include "testsuite.ycp";

    map READ = $[
	"target" : $[ "size": -1, "tmpdir" : "/tmp", ]
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

    define void TEST_It_Now(list detected) ``{
	Tv::detected_cards = detected;
	DUMP("---------------");

	// Tv::cards (... already installed cards) == nil
	Tv::cards = nil;
	list detected_items = Tv::DetectedCardsAsItems();
	DUMP(detected_items);

	// Tv::cards == $[]
	Tv::cards = [];
	detected_items = Tv::DetectedCardsAsItems();
	DUMP(detected_items);

	// Tv::cards == card1
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ]
	    ];
	detected_items = Tv::DetectedCardsAsItems();
	DUMP(detected_items);

	// Tv::cards == card1, nil, nil, card4
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ],
	      nil,
	      nil,
	      $[ "module" : "xyz",
		 "parameters" : $[ "par_other" : "testvalue2" ],
		 "name" : "My TV_card",
		 "unique_key" : "j7F3.J8dhgs34fK2",
		 "sound_card_no" : 1
	      ]
	    ];
	detected_items = Tv::DetectedCardsAsItems();
	DUMP(detected_items);

	}
    // testing with different values of detected_cards
    TEST_It_Now(nil);

    TEST_It_Now([ $[ "module" : "bttv",
		 "name" : "First detected card",
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0 ]]);

	TEST_It_Now([ $[ "module" : "xyz",
		 "name" : "My TV_card",
		 "unique_key" : "j7F3.J8dhgs34fK2",
	      ],
				  $[ "module" : "bttv",
		 "name" : "Some other detected card",
		 "unique_key" : "hluk.YYY2",
		 "sound_card_no" : 0 ]]);

}
