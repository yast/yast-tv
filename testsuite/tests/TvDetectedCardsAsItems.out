Read	.target.size "/etc/install.inf" nil
Read	.etc.install_inf.Sourcemounted nil
Read	.etc.install_inf.AutoYaST nil
Read	.etc.install_inf.Demo nil
Read	.etc.install_inf.Textmode nil
Read	.etc.install_inf.Manual nil
Read	.etc.install_inf.Console nil
Read	.etc.install_inf.Braille nil
Read	.etc.install_inf.VNC nil
Read	.etc.install_inf.UseSSH nil
Read	.etc.install_inf.Locale nil
Read	.etc.install_inf.InstMode nil
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system []
Dump	---------------
Dump	[`item (`id (-1), "Other (not detected)", true)]
Dump	[`item (`id (-1), "Other (not detected)", true)]
Dump	[`item (`id (-1), "Other (not detected)", true)]
Dump	[`item (`id (-1), "Other (not detected)", true)]
Dump	---------------
Dump	[`item (`id (0), "First detected card", true), `item (`id (-1), "Other (not detected)", false)]
Dump	[`item (`id (0), "First detected card", true), `item (`id (-1), "Other (not detected)", false)]
Dump	[`item (`id (-1), "Other (not detected)", true)]
Dump	[`item (`id (-1), "Other (not detected)", true)]
Dump	---------------
Dump	[`item (`id (0), "My TV_card", true), `item (`id (1), "Some other detected card", false), `item (`id (-1), "Other (not detected)", false)]
Dump	[`item (`id (0), "My TV_card", true), `item (`id (1), "Some other detected card", false), `item (`id (-1), "Other (not detected)", false)]
Dump	[`item (`id (0), "My TV_card", true), `item (`id (1), "Some other detected card", false), `item (`id (-1), "Other (not detected)", false)]
Dump	[`item (`id (1), "Some other detected card", true), `item (`id (-1), "Other (not detected)", false)]
